azavea/raster-foundry
# Add model tags to database

## Overview

This commit adds support for inserting model tags in database. A
migration adds the necessary table and constraints and initial datamodel
and database table is added in the scala code.
### Checklist


## Testing Instructions

 * Start the VM and get the development environment set up
 * Load development data if you have not yet done so (`./scripts/load_development_data`)
 * Start an SBT console and run migrations
```./scripts/console app-server ./sbt
    mg update
    mg apply
 ```
 * Open a scala shell
 ```scala
    project app
    console
```
 * Verify that the following snippet creates a tag, retrieves it, and throws an exception when trying to insert a duplicate tag
    ```scala
    // User ID after migrations
    val userId = "auth0|57b744b09da7f9936f14fd5b"
    val modelTagCreate = ModelTag.Create(publicOrgId, "Testing")
    val createdFuture = ModelTags.insertModelTag(modelTagCreate, userId)
    val created = Await.result(createdFuture, 10 seconds)
    val retrievedFuture = ModelTags.getModelTag(created.id)
    val retrieved = Await.result(retrievedFuture, 10 seconds)

    val duplicateModelTag = ModelTags.insertModelTag(modelTagCreate, userId)

    // Should throw an exception
    Await.result(duplicateModelTag, 10 seconds)
    ```

Closes #631
